src/solve.jl

```diff
@@ -296,8 +296,35 @@ function init(prob::AbstractDDEProblem{uType,tType,lType,isinplace}, alg::algTyp
 
     # set up additional initial values of newly created DDE integrator
     # (such as fsalfirst) and its callbacks
-    initialize!(dde_int)
-    initialize!(integrator.opts.callback, integrator.t, u, dde_int)
+
+    if initialize_integrator
+      integrator.u_modified = true
```

Comment:
I think we should (mostly) only operate on `dde_int` here and in the following lines. `integrator` is the wrong integrator, callbacks are initialized with `dde_int`.